packages/samples/headless-react/src/components/standalone-search-box/standalone-search-box.fn.tsx

```diff
@@ -0,0 +1,73 @@
+import {useEffect, useState, FunctionComponent} from 'react';
+import {StandaloneSearchBox as HeadlessStandaloneSearchBox} from '@coveo/headless';
+
+interface StandaloneSearchBoxProps {
+  controller: HeadlessStandaloneSearchBox;
+}
+
+export const StandaloneSearchBox: FunctionComponent<StandaloneSearchBoxProps> = (
+  props
+) => {
+  const {controller} = props;
+  const [state, setState] = useState(controller.state);
+
+  useEffect(() => controller.subscribe(() => setState(controller.state)), []);
+
+  function getURIParameters() {
+    const {redirectTo, value} = state;
+    const query = encodeURIComponent(value);
+    // The query pipeline can trigger different redirection URLs, as documented here: https://docs.coveo.com/en/1458/tune-relevance/trigger-query-pipeline-feature
+    // Unless a redirect trigger is added, you won't need such conditions. These redirections are only verified here as an example.
+    if (redirectTo === 'https://mywebsite.com/search') {
+      return '/' + query;
+    } else if (redirectTo === 'https://www.google.com') {
+      return '?q=' + query;
+    }
+    return '?search=' + query;
```

Comment:
Why are we checking the `redirectTo` url in the sample? Why not just use the `redirectTo` in state?

I really believe the goal with these samples should be to show a simple use-case with the controller.

When a sample is simple, people will think of headless as simple. When a sample is complex, headless will be perceived as complicated, which would not be a fair representation.

Reply:
I'm checking it for the reason I wrote above.

However, it's not strictly necessary and I agree, it does make the sample look more complicated than necessary. I can remove it